esp-hal/src/dma/mod.rs

```diff
@@ -1348,6 +1350,10 @@ where
         }
     }
 
+    fn stop_transfer(&mut self) {
+        R::reset_in();
```

Comment:
Typo?

Reply:
Nah it's general confusion after wrestling with this thing for far too long.

Reply:
Actually, what's wrong with resetting the in direction here?

Reply:
To someone who's familiar with the DMA hardare, the method name is confusing. I would've assumed this was setting the stop bit in the inlink register (like in my I8080 move api PR) rather than resetting the state machine.

Reply:
Fair. Well, this PR entered the stage where what I want seems possible, and we can think about exactly how to do things, so I'll revisit if this is even necessary or we're fine with just cutting back the SPI datalength to 0-ish